### ISSUE TITLE
Sample weigh support for regression problems

### ISSUE THREAD

**Issue by kmedved** [dreamquark-ai/tabnet#132]
## Feature request

**What is the expected behavior?**
It would be very helpful to add sample weight support for regression problems. The idea would be to add a 'sample_weight' parameter to the .fit() call, and give a weighted regression.

**What is motivation or use case for adding/changing the behavior?**
Many datasets involve different sample weights. This is especially common with sports data (where I work), but is frequently used elsewhere.

**How should this be implemented in your opinion?**
The usual implementation I've seen has been to multiply the individual residuals by the sample weight, but I am not very familiar with the underlying math here, so don't know how it would work.

**Are you willing to work on this yourself?**
I am happy to help, but my understanding of the underlying code is lacking at the moment.

**Comment by Optimox** [comment]
Hi @kmedved,

This is a good feature to add.

Basically we have two options to deal with training weights, either by changing the weights in the loss or changing the sampling weights in dataloaders to do over/under sampling.

I'll have a look and should come back with a concrete solution pretty soon!

**Comment by kmedved** [comment]
Thanks for the quick work on this @Optimox.

**Comment by kmedved** [comment]
Would it be possible to add documentation on this feature? I'm struggling to understand how to pass both training and validation weights. I can call a `.fit() `with a weights parameter, but I'm not sure how to separately designate the validation weights in particular. For instance, LightGBM has a parameter for` 'eval_sample_weight'` if an `eval_set `is passed, but I can't find anything equivalent here.

Thanks again.

**Comment by Optimox** [comment]
@kmedved sorry I missed your question, you can't set specific weights for validation at the moment...

**Comment by kmedved** [comment]
Got it. Thanks @Optimox. That would be helpful to add, since if we have weights on the training data, we're naturally going to end up with them on the validation data too.

Appreciate all the work on this library.
